src/services/dashboards/resource.js

```diff
@@ -330,6 +331,438 @@ const switchToTopicCentric = (input) => {
+      FROM ${tblNames.createdFlatResourceTempTableName} tf
+      GROUP BY url, title, "rollUpDate"
+      ORDER BY "url", tf."rollUpDate" ASC),
+    distincturls AS (
+    SELECT DISTINCT url, title
+    FROM ${tblNames.createdFlatResourceTempTableName}
+    ),
+    totals AS
+    (
+      SELECT
+      url,
+      title,
+      SUM("resourceCount") AS "totalCount"
+      FROM urlvals
+      GROUP BY url, title
+      ORDER BY SUM("resourceCount")  DESC,
+      url ASC
+      LIMIT 10
+    ),
+    series AS
+    (
+      SELECT * FROM ${tblNames.createdFlatResourceHeadersTempTableName}
+    )
+      SELECT
+      d.url,
+      d.title,
+      to_char(s."date", 'Mon-YY') AS "rollUpDate",
+      s."date",
+      coalesce(u."resourceCount", 0) AS "resourceCount",
+      t."totalCount"
+      FROM distincturls d
+      JOIN series s
+        ON 1=1
+      JOIN totals t
+        ON d.url = t.url
+      LEFT JOIN urlvals u
+        ON d.url = u.url AND to_char(s."date", 'Mon-YY') = u."rollUpDate"
+      ORDER BY 1,4 ASC;
+  `;
+
+  return sequelize.query(
+    resourceUseSql,
+    {
+      type: QueryTypes.SELECT,
+      transaction,
+    },
+  );
+}
+
+function getTopicsUseSql(tblNames, transaction) {
+  const topicUseSql = `
+  WITH topics AS (
+      SELECT
+        f.id,
+        t.name,
+        f."rollUpDate",
+        count(DISTINCT f.url) AS "resourceCount" -- Only count each resource once per ar and topic.
+      FROM ${tblNames.createdTopicsTempTableName}  t
+        JOIN ${tblNames.createdAroTopicsTempTableName} arot
+        ON t.id = arot."topicId"
+        JOIN ${tblNames.createdFlatResourceTempTableName} f
+        ON arot."activityReportId" = f.id
+        GROUP BY f.id, t.name, f."rollUpDate"
+        ORDER BY t.name, f."rollUpDate" ASC
+    ),
+    topicsperdate AS
+    (
+    SELECT
+      "name",
+      "rollUpDate",
+      SUM("resourceCount") AS "resourceCount"
+    FROM topics
+    GROUP BY "name", "rollUpDate"
+    ),
+    totals AS
+    (
+      SELECT
+        name,
+        SUM("resourceCount") AS "totalCount"
+      FROM topicsperdate
+      GROUP BY name
+      ORDER BY SUM("resourceCount") DESC
+    ),
+    series AS
+    (
+      SELECT * FROM ${tblNames.createdFlatResourceHeadersTempTableName}
+    )
+    SELECT
+      d.name,
+      to_char(s."date", 'Mon-YY') AS "rollUpDate",
+      s."date",
+      coalesce(t."resourceCount", 0) AS "resourceCount",
+      tt."totalCount"
+    FROM ${tblNames.createdTopicsTempTableName} d
+    JOIN series s
+      ON 1=1
+    JOIN totals tt
+      ON d.name = tt.name
+    LEFT JOIN topicsperdate t
+      ON d.name = t.name AND to_char(s."date", 'Mon-YY') = t."rollUpDate"
+    ORDER BY 1, 3 ASC;`;
+  return sequelize.query(
+    topicUseSql,
+    {
+      type: QueryTypes.SELECT,
+      transaction,
+    },
+  );
+}
+
+function getOverview(tblNames, totalReportCount, transaction) {
+  // - Number of Participants -
+  const numberOfParticipants = sequelize.query(`
```

Comment:
please add the /* sql */ at the start of the query string to allow for syntax highlighting to make the code easier to read.